lib/auth/grpcserver.go

```diff
@@ -2929,14 +2930,32 @@ func (g *GRPCServer) GetToken(ctx context.Context, req *types.ResourceRequest) (
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	provisionTokenV2, ok := t.(*types.ProvisionTokenV2)
-	if !ok {
-		return nil, trace.Errorf("encountered unexpected token type: %T", t)
+
+	v2, err := t.V2()
```

Comment:
This might become problematic if users receive a v2 token converted from an original v3 token stored in auth server. 
If they try to update as v2, they can replace the original token with the v2 version